### DISCUSSION TITLE
AutoCompleteBox doesn't show pick list when inside DataGrid

### DISCUSSION THREAD

**Discussion by AArnott** [AvaloniaUI/Avalonia#13855] [Q&A]
I have a few uses of `AutoCompleteBox` in my app. It works great, except when it appears inside a `DataGrid` via `DataGridTemplateColumn.CellEditingTemplate`. There, it behaves just like a `TextBox`, showing no picklist from the `ItemsSource` collection, and never setting the `SelectedItem` bound property.

Is this a known issue? Any suggestions?

Thanks.

**Comment by sccombp** [comment]
Do you need to put it in the CellEditingTemplate? I use AutoCompleteBoxes all the time inside datagrids and they work just fine for editing if you put them inside the CellTemplate

heres an example from one of my projects
```xml
<DataGridTemplateColumn Header="Action" CanUserSort="True" SortMemberPath="SelectedAction">
  <DataGridTemplateColumn.CellTemplate>
    <DataTemplate>
      <AutoCompleteBox
            ItemsSource="{Binding AssessmentActions}"
            ValueMemberBinding="{Binding Description}"
            IsTextCompletionEnabled="True"
            FilterMode="Contains"
            Text="{Binding SelectedAction, Mode=TwoWay}"
            ToolTip.Tip="{Binding Path=$self.Text, Mode=OneWay}">
      </AutoCompleteBox>
    </DataTemplate>
  </DataGridTemplateColumn.CellTemplate>
</DataGridTemplateColumn>
```

Here it is in action
![image](https://github.com/AvaloniaUI/Avalonia/assets/124018297/851f372c-6741-48e8-b950-fb1cd0160eaa)

**Reply by AArnott** [reply]
The difference between CellTemplate and CellEditingTemplate is whether the user is expecting to make changes. CellTemplate isn't supposed to be editable, so putting an AutoCompleteBox there seems wrong.
That said, I'll try it to see if the results are any different.

**Reply by timunie** [reply]
depends on your UI design imo.

**Comment by timunie** [comment]
@AArnott without any code, not much we can do. Maybe wrong DataContext for the ItemsSource?

**Reply by sccombp** [reply]
@AArnott Tim just jogged my memory, when binding an AutoCompleteBox in a datagrid you are binding at the rows object level and not the view model. The ItemsSource for the AutoCompleteBox needs to come from the row object.

try setting ```x:CompileBindings="True"``` in your view and see if the compiler kicks out any errors

**Reply by timunie** [reply]
@sccombp you can also bind to your parents ViewModel with named snytax. `#root.((vm:MainViewModel)DataContext).MyList`

**Reply by sccombp** [reply]
@timunie I owe you a tasty beverage, it never occurred to me you could get to the DataContext that way.

**Reply by thevortexcloud** [reply]
You also don't need to enable CompiledBindings globally to test it. You can just use the `CompiledBinding` markup extension.

**Reply by AArnott** [reply]
Thanks. I'll share some code tomorrow. I'm very much expecting the data binding to be against the viewmodel behind the _row_ rather than the whole page/window. In fact the data binding works at the TextBox behavior level. It's just the pop up list that doesn't.

**Reply by timunie** [reply]
>  In fact the data binding works at the TextBox behavior level. It's just the pop up list that doesn't.

I still think the Items may be defined in the parent VM while the test is bound in ItemVM. That would explain it,

**Reply by AArnott** [reply]
I created a standalone app to demonstrate the problem, but it worked as it should. I then went back to the original app and tried again, and it too worked as it should. So... I guess nothing to see here any more. No idea why it wasn't working before, as I haven't changed any relevant code.

Thank you for your replies and ideas to help!
